### PR TITLE
Add new function to load and run pipeline

### PR DESCRIPTION
**Pull request by Reskov**
Split run_pipeline function on two: 
- load_and_run_pipeline
- run_pipeline
The main reason is to add capability to call pipeline context directly. See example of use case inside issue #80

### CODE REVIEW

**Comment by Reskov** [comment]
Test are passed locally for python 3.7, but flake8 failed in context.py, which I haven't touch
==== 455 passed in 5.80 seconds ====
./pypyr/context.py:578:39: W504 line break after binary operator
./pypyr/context.py:582:39: W504 line break after binary operator
./pypyr/steps/assert.py:37:26: W504 line break after binary operator

File: tests/unit/pypyr/pipelinerunner_test.py [hunk 1]
```diff
@@ -384,22 +388,23 @@ def test_run_pipeline_steps_error_no_context(mocked_work_dir,
...
 
-    pypyr.pipelinerunner.run_pipeline(
+    pypyr.pipelinerunner.load_and_run_pipeline(
         pipeline_name='arb pipe',
         pipeline_context_input='arb context input',
         working_dir='arb/dir',
         context=existing_context)
 
-    existing_context.working_dir == 'from/context'
+    assert existing_context.working_dir == 'from/context'
```
Comment (345thomas) [inline tests/unit/pypyr/pipelinerunner_test.py]: great catch!

File: tests/unit/pypyr/steps/safeshell_test.py [hunk 2]
```diff
@@ -40,8 +41,9 @@ def test_shell_sequence_with_string_interpolation():
 
 def test_shell_error_throws():
     """Shell process returning 1 should throw CalledProcessError"""
+    cmd = '/bin/false' if platform.system() != 'Darwin' else '/usr/bin/false'
```
Comment (345thomas) [inline tests/unit/pypyr/steps/safeshell_test.py]: thanks!

**Comment by 345thomas** [comment]
Sorry about the flake8 breakages - seems the default selector for flake8 was picking two mutually exclusive rules (both break before AND break after binary operators) as of the last October update). . . more info here https://github.com/pypyr/pypyr-cli/pull/82

I fixed and merged the underlying master branch - as expected, nothing wrong with your code!

**Pull request merged**